src/compiler/checker.ts

```diff
@@ -17731,6 +17735,20 @@ namespace ts {
             return type;
         }
 
+        function checkParenthesizedExpression(node: ParenthesizedExpression, checkMode?: CheckMode): Type {
+            if (isInJavaScriptFile(node)) {
+                if (node.jsDoc) {
+                    const typecasts = flatten<JSDocTag>(map(node.jsDoc, doc => filter(doc.tags, tag => tag.kind === SyntaxKind.JSDocTypeTag)));

```

Comment:
can you use `flatMap` instead? `filter` should always return an array so you don't need the `if (isArray ...` check in `flatten`.